### PR TITLE
174 update cache for run

### LINKED ISSUE #174: update cache to add duration and list of paths

### PR DESCRIPTION
**Pull request by HoLuDyEr**
This PR updates the cache to store the refromatted runs.

closes #174

### CODE REVIEW

File: app/src/androidTest/java/com/epfl/drawyourpath/userProfile/cache/UserModelCachedTest.kt
Comment (HugoHof) [inline app/src/androidTest/java/com/epfl/drawyourpath/userProfile/cache/UserModelCachedTest.kt:1]: Remark: certain cache functions test function like fun addNewRunAddRunAndModifyProgress() (line 214) and assertEquals user contains commented code

**Review by HugoHof — CHANGES_REQUESTED** [review]
Thank you to adapt the cache for the new run and path. You also clean and restructure certain part of your code. That is a very good PR. Thank you for your work. There is just few remarks and we are good. 
I don't remember, will you store the score Gove by the ML in your new task ?

File: app/src/androidTest/java/com/epfl/drawyourpath/userProfile/cache/UserModelCachedTest.kt
Comment (HoLuDyEr) [reply app/src/androidTest/java/com/epfl/drawyourpath/userProfile/cache/UserModelCachedTest.kt:1]: I deleted commented code inisde assertEquals as you said the achievements have been deleted.
For the run test, I will fix this test in another PR.

**Comment by HugoHof** [comment]
> > Thank you to adapt the cache for the new run and path. You also clean and restructure certain part of your code. That is a very good PR. Thank you for your work. There is just few remarks and we are good. I don't remember, will you store the score Gove by the ML in your new task ?
> 
> Yes, I have split my task in 2 so that I could finish this part more quickly to have a functional cache. I will add the ml score and shape and also fix some test with the run in another task.

Thank you for your change and you clarification.
You can merge.

**Review by HugoHof — APPROVED** [review]

**Pull request merged**